Sources/NIOCore/NIOOnLoopSendable.swift

```diff
@@ -0,0 +1,78 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+public struct NIOOnLoopSendable<Value>: @unchecked Sendable {
```

Comment:
I wonder if instead of `NIOOnLoopSendable` we should say `NIOLoopBoundValue`. How does that sound?

Reply:
I do like LoopBound(Value), probably even more without "Value", because `NIOLoopBound<ChannelHandler>` sounds like actual English :)

Reply:
The counterpart `NIOLoopBoundBox<ChannelHandler>` isn't ideal but we could opt for `NIOLoopBoundBoxed<ChannelHandler>`

Reply:
Yeah, so I proposed `NIOLoopBoundValue` in part because it clarifies the (potential) value semantics of this type, while `NIOLoopBoundBox` is always referential. Could also be `NIOLoopBoundReference<>`. But I'm happy enough to drop `Value`.

Reply:
I'd be in favour of dropping because that's kinda the default (like Array/Optional/... which are equally potentially fully value typed). I'm ambivalent about if we use Box or Reference. I went for Box because of NIOLockedValueBox.

Reply:
@Lukasa renamed to `NIOBoundValue(Box)`, lmkwyt